index.spec.ts

```diff
@@ -51,6 +51,45 @@ test("arithmetic example", () => {
   expect(toIPv4(fromIPv4("3.5.22.255") + 22)).toBe("3.5.23.21");
 });
 
+/*
+  IPv4 Additional Testing 
+*/
+
+test("should be 0.0.0.0", () => {
+  expect(toIPv4(0)).toBe("0.0.0.0");
+});
+
+test("should be 0", () => {
+  expect(fromIPv4("0.0.0.0")).toBe(0);
+});
+
+test("should throw error for invalid IPv4 format", () => {
+  expect(() => fromIPv4("999.999.999.999")).toThrow();
+  expect(() => fromIPv4("abcd")).toThrow();
+  expect(() => fromIPv4("192.168.0.256")).toThrow();
+  expect(() => fromIPv4("192.168.-1.1")).toThrow();
+});
+
+test("should handle IPv4 with leading zeros", () => {
+  expect(fromIPv4("192.168.001.001")).toBe(fromIPv4("192.168.1.1"));
+  expect(toIPv4(fromIPv4("001.002.003.004"))).toBe("1.2.3.4");
+});
+
+test("should throw error for numbers beyond IPv4 range", () => {
+  expect(() => toIPv4(4294967296)).toThrow(); // beyond 32-bit range
+  expect(() => toIPv4(BigInt(4294967296))).toThrow();
```

Comment:
This line is the problem. You give `toIPv4()` a `bigint` but `number` is expected. Like in the line above. If you delete this line, it will work.

Reply:
I removed those lines from the test file has mentioned .